pkgs/build-support/docker/make-layers.nix

```
@@ -0,0 +1,47 @@
+{
+  coreutils,
+  flattenReferencesGraph,
+  lib,
+  jq,
+  stdenvNoCC,
+}:
+{
+  closureRoots,
+  excludePaths ? [ ],
+  # This could be a path to (or a derivation producing a path to)
+  # a json file containing the pipeline
+  pipeline ? [ ],
+  debug ? false,
+}:
+if closureRoots == [ ] then
+  builtins.toFile "docker-layers-empty" "[]"
+else
+  stdenvNoCC.mkDerivation {
+    name = "docker-layers";
+    __structuredAttrs = true;
+    # graph, exclude_paths and pipeline are expected by the
+    # flatten_references_graph executable.
+    exportReferencesGraph.graph = closureRoots;
+    exclude_paths = excludePaths;
+    inherit pipeline;
+    # builder cannot refer to derivation outputs
+    PATH = "${coreutils}/bin:${flattenReferencesGraph}/bin:${jq}/bin";
```

```suggestion
    PATH = lib.makeBinPath [ buildPackages.coreutils buildPackages.flattenReferencesGraph buildPackages.jq ];
```

---

Might be better to just use the stdenv `builder` and have a `buildCommand` (which is what `runCommand` does). Then you can use `nativeBuildInputs`. No need to use `buildPackages` in that case.
Otherwise:


```suggestion
    nativeBuildInputs = [ coreutils flattenReferencesGraph jq ]";
```

---

done in 266283bec3ae